types/experiments.ts

```diff
@@ -0,0 +1,81 @@
+/** All fields are publically accessible via the API and available to clients */
+interface Experiment {
+  /** Unique identifier for the experiment */
+  slug: string;
+  /** Publically-accesible name of the experiment */
+  name: string;
+  /** Short public description of the experiment */
+  public_description: string;
+
+  /** Current state of the experiment. */
+  status: "Draft" | "Review" | "Accepted" | "Rejected" | "Live" | "Complete";
+  /** Is the experiment currently live in production? i.e., published to remote settings? */
+  isPublished: boolean;
+  /** Are we continuing to enroll new users into the experiment? */
+  isEnrollmentPaused: boolean;
+
+  /** Reference to an Audience, which contains targeting information */
+  audience: Audience;
+  /** Bucketing configuration */
+  bucketConfig: BucketConfig;
+  /** A list of cusom feature metrics relevant to the experiment analysis */
+  featureMetrics: Array<FeatureMetric>;
+  /** Branch configuration for the experiment */
+  branches: Array<Branch>;
+
+  /** Experimenter URL. Generated by Experimenter. */
+  experiment_url: string;
+  /** Actual publish date of the experiment */
+  start_date: Date;
+  /** Actual end date of the experiment */
+  end_date: Date;
+}
+
+/** These fields are internal to Experimenter */
+interface ExperimentDetails extends Experiment {
+  hypothesis: string;
+  namespace?: string;
+  proposed_start_date: Date;
+  proposed_end_date: Date;
+  proposed_enrollment: number;
+}
+
+interface BucketConfig {
+  randomization_unit: "client_id";
+  inputs: Array<string>
```

Comment:
If this is mimicking Normandy's bucket strategy, I think that this should be more concrete and be `namespace: string`. In practice, all of the Normandy recipes that use bucket sample filters have exactly two inputs, the randomization unit (`normandy.userId`) and the namespace. There are a few exceptions to that, but almost all of them are errors. Additionally, any extra inputs we could need can be mimicked by appending more data to the namespace string.

Reply:
yep, that sounds good to me!